tests/parametric/test_otel_env_var.py

```diff
@@ -0,0 +1,233 @@
+import pytest
+from utils import missing_feature, context, scenarios
+
+otel_env_var_to_internal_name_mapper = {
+    "nodejs": {
+        "OTEL_SERVICE_NAME": "service",
+        "OTEL_LOG_LEVEL": "logLevel",
+        "OTEL_TRACES_SAMPLER": "sampleRate",
+        #'OTEL_TRACES_EXPORTER' not exposed by the tracer
+        "OTEL_METRICS_EXPORTER": "runtimeMetrics",
+        "OTEL_RESOURCE_ATTRIBUTES": "tags",
+        # both tracePropagationStyle.inject & tracePropagationStyle.extract set the value of OTEL_PROPAGATORS, so test either
+        "OTEL_PROPAGATORS": "tracePropagationStyle.inject",
+        "OTEL_VERSION": "version",
+        "OTEL_ENV": "env",
+    }
+}
+
+
+def get_nested_dict_value(data, dot_key):
+    keys = dot_key.split(".")
+    for key in keys:
+        if isinstance(data, dict):
+            data = data.get(key)
+        else:
+            return None
+    return data
+
+
+def get_config_value(config, language, env_var):
+    return get_nested_dict_value(config, otel_env_var_to_internal_name_mapper[language][env_var])
+
+
+@scenarios.parametric
+class Test_Otel_Env_Var:
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library < "nodejs@5.11.0", reason="Implemented in v5.11.0, v4.35.0, &v3.56.0")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                "DD_SERVICE": "service",
+                "OTEL_SERVICE_NAME": "otel_service",
+                "DD_TRACE_LOG_LEVEL": "error",
+                "OTEL_LOG_LEVEL": "debug",
+                "DD_TRACE_SAMPLE_RATE": "0.5",
+                "OTEL_TRACES_SAMPLER": "traceidratio",
+                "OTEL_TRACES_SAMPLER_ARG": "0.1",
+                "DD_TRACE_ENABLED": "true",
```

Comment:
Is there a corresponding assert for DD_TRACE_ENABLED?

Reply:
i can add one as a separate test because for node dd_trace_enabled is not exposed in the config object